test-suite/handwritten-src/ts/AsyncTest.ts

```diff
@@ -65,6 +65,17 @@ class AsyncTest extends TestCase {
         const s = await this.m.testsuite.TestHelpers.checkAsyncComposition(new AsyncInterfaceImpl());
         assertEq(s, "42");
     }
+
+    async testEarlyThrow() {
+        const r = this.m.testsuite.TestHelpers.asyncEarlyThrow();
+        let s = null;
+        try {
+            await r;
```

Comment:
@bkotsopoulossc 
Exception is thrown here in `await` rather than in the async call above.

Reply:
Amazing